src/npctalk_funcs.cpp

```diff
@@ -736,6 +737,27 @@ void talk_function::buy_100_logs( npc &p )
     add_msg( m_good, _( "%s drops the logs off in the garage…" ), p.name );
 }
 
+/*
+ * Function to make the npc drop non favorite items at their current position.
+ */
+void talk_function::drop_items_in_place( npc &p )
+{
+    std::vector<drop_or_stash_item_info> to_drop;
+
+    // add all non favorite items to the drop off list
+    for( const item_location &npcs_item : p.all_items_loc() ) {
+        if( !npcs_item->is_favorite ) {
```

Comment:
```suggestion
    // add all non favorite carried items to the drop off list
    for( const item_location &npcs_item : p.all_items_loc() ) {
        if( npcs_item.where() != item_location::type::character && !npcs_item->is_favorite ) {
```

Reply:
to fix "dropping all their clothes" (and weapon)

Reply:
thanks for the suggestion. So i guess type::character referrs to worn or wielded? (items the character has equipped currently)?

Reply:
Right, you can infer that from reading `Character::all_items_loc` and the 2 `item_location` constructors that it uses (one delegates to item_on_person, the other to item_in_container). I wish it was documented a little bit more explicitly.